src/libraries/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbConnectionPoolIdentity.cs

```diff
@@ -30,7 +32,7 @@ internal static DbConnectionPoolIdentity GetCurrent()
             throw new PlatformNotSupportedException();
         }
 
-        public override bool Equals(object? value)
+        public override bool Equals([NotNullWhen(true)] object? value)
```

Comment:
This doesn't hold when `this == NoIdentity` - the method will return true even if value is false. That definitely seems like questionable behavior, but I'm not sure there's an actual bug here and this old legacy code (AFAIK only used as a support type for the System.Data.{Odbc,OleDb}).

Reply:
Reverted